std/math.d

```diff
@@ -7607,3 +7607,35 @@ T truncPow2(T)(const T val) if (isFloatingPoint!T)
         assert(truncPow2(T.init).isNaN);
     }
 }
+
+// Credit: Matthias Bentrup
+/**
+Params:
+    x = the number to test
+
+Returns:
+    `true` if `x` is a nonzero power of two.
+*/
+@safe @nogc nothrow pure
+bool isPowerOf2(T)(T x) if (isUnsigned!T)
+{
+    return (x & -x) > (x - 1);
+}
+
+///
+@safe @nogc nothrow pure
+unittest
+{
+    assert(!isPowerOf2(0));
```

Comment:
you need to postfix the literals with 'U'. Maybe dups all the tests, first battery with U, second with UL.